sql/sql_db.cc

```diff
@@ -890,16 +874,38 @@ mysql_rm_db_internal(THD *thd, const LEX_CSTRING *db, bool if_exists, bool silen
     else
     {
       push_warning_printf(thd, Sql_condition::WARN_LEVEL_NOTE,
-			  ER_DB_DROP_EXISTS, ER_THD(thd, ER_DB_DROP_EXISTS),
+        ER_DB_DROP_EXISTS, ER_THD(thd, ER_DB_DROP_EXISTS),
                           db->str);
       error= false;
       goto update_binlog;
     }
   }
 
-  if (find_db_tables_and_rm_known_files(thd, dirp, dbnorm, db_path, &tables))
+  del_dbopt(opt_path); // Remove dboption hash entry
```

Comment:
What's the effect of moving this after the exists check?

Reply:
We can't delete the db before calling `exists`.
The `del_dbopt` call is not important here and is just logically grouped together with `myrocks::rocksdb_db_delete(opt_path)`